src/graph/graph.c

```diff
@@ -1288,22 +1376,40 @@ RG_Matrix Graph_GetRelationMatrix
 	ASSERT(relation_idx == GRAPH_NO_RELATION ||
 		   relation_idx < Graph_RelationTypeCount(g));
 
-	RG_Matrix m = GrB_NULL;
+	Delta_Matrix m = GrB_NULL;
 
 	if(relation_idx == GRAPH_NO_RELATION) {
 		m = g->adjacency_matrix;
 	} else {
-		m = g->relations[relation_idx];
+		m = g->relations[relation_idx].R;
 	}
 
-	g->SynchronizeMatrix(g, m);
+	size_t n = Graph_RequiredMatrixDim(g);
+	g->SynchronizeMatrix(g, m, n, n);
 
-	if(transposed) m = RG_Matrix_getTranspose(m);
+	if(transposed) m = Delta_Matrix_getTranspose(m);
 
 	return m;
 }
 
-RG_Matrix Graph_GetAdjacencyMatrix
+Delta_Matrix Graph_MultiEdgeRelationMatrix
+(
+	const Graph *g,
+	RelationID relation_idx
+) {
+	ASSERT(g);
+	ASSERT(relation_idx != GRAPH_NO_RELATION &&
+		   relation_idx < Graph_RelationTypeCount(g));
+
+	Delta_Matrix m = g->relations[relation_idx].ME;
+
+	size_t edge_cap = g->edges->itemCap;
+	g->SynchronizeMatrix(g, m, edge_cap, edge_cap);
```

Comment:
the multi-edge matrix should be dimensions `NxM` where `N = ceil(edge_cap / 2)` and `M = edge_cap`

Reply:
if all the edges are connecting the same src and dst?

Reply:
the matrix is hypersparse no much memeory overhead